### PR TITLE
samples: nrf_desktop: Switch to new charging HW

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-482

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

**Comment by pdunaj** [comment]
tested on new pca20041 - charging detection works like a charm
I still need to check battery error

**Comment by pdunaj** [comment]
I tested error case by removing the battery while charging - got crash as it seems we had a bug in our code for some time (case when battery error was indicated was not handled correctly). I have added a fix for this issue to this PR.
Now error and charging states are correctly indicated.

**Comment by pdunaj** [comment]
Hi @wiminordic , can you tell what is wrong with the test? It looks like connection issue. That would be unrelated to the change.

**Comment by wiminordic** [comment]
This is a known minor issue: DESK-476

**Comment by pdunaj** [comment]
@wiminordic so we can merge?

File: samples/nrf_desktop/src/hw_interface/Kconfig [hunk 1]
```diff
@@ -219,6 +219,21 @@ config DESKTOP_BATTERY_CSO_PIN
...
 
+choice
+	prompt "Select CSO pin pull"
+	default DESKTOP_BATTERY_CSO_PULL_NONE
+
+config DESKTOP_BATTERY_CSO_PULL_NONE
+	bool "Disable CSO pin pull"
+
+config DESKTOP_BATTERY_CSO_PULL_UP
+	bool "Enable CPI pin pull up"
```
Comment (Qbicz) [inline samples/nrf_desktop/src/hw_interface/Kconfig]: s/CPI/CSO

File: samples/nrf_desktop/configuration/nrf52840_pca20041/app_ZRelease.conf [hunk 2]
```diff
@@ -21,7 +21,9 @@ CONFIG_DESKTOP_LED_COLOR_COUNT=3
 CONFIG_DESKTOP_BATTERY_DISCRETE=y
 CONFIG_DESKTOP_BATTERY_MONITOR_ENABLE_PIN=6
 CONFIG_DESKTOP_BATTERY_CSO_PIN=8
-CONFIG_DESKTOP_BATTERY_CSO_FREQ=1000
+CONFIG_DESKTOP_BATTERY_CSO_PULL_UP=y
+CONFIG_DESKTOP_BATTERY_CSO_FREQ=1
+
```
Comment (MarekPieta) [inline samples/nrf_desktop/configuration/nrf52840_pca20041/app_ZRelease.conf]: excessive empty line ?

File: samples/nrf_desktop/src/hw_interface/Kconfig (hunk 1 above)
Comment (pdunaj) [reply samples/nrf_desktop/src/hw_interface/Kconfig]: Fixed

File: samples/nrf_desktop/configuration/nrf52840_pca20041/app_ZRelease.conf (hunk 2 above)
Comment (pdunaj) [reply samples/nrf_desktop/configuration/nrf52840_pca20041/app_ZRelease.conf]: Fixed

**Comment by pdunaj** [comment]
Hi @pizi-nordic , could you merge?

**Pull request merged**